### DISCUSSION TITLE
Packaging the CLI for openSUSE: is there a testsuite or test command for the CLI?

### DISCUSSION THREAD

**Discussion by kastl-ars** [k8up-io/k8up#1010] [Q&A] [answered]
Hi all,

I just came across k8up and found it interesting, so I packaged the CLI for openSUSE. Before submitting it to the distribution, I wanted to make sure the executable is really working.

I'll prepare a vagrant-based setup with k8up installed in Kubernetes and will play around with this.

But, for the packaging, is there a testsuite or a test command that I can run to make sure the executable is working? Other than using `k8up --version` and grep'ing for the correct version/commit/date?

Kind Regards,
Johannes

**Comment by Kidswiss** [comment, marked as answer]
Hi @kastl-ars 

Currently, we only have test suites for the operator itself. However the restores in these currently rely on some fairly old bash scripts to create the restore objects manually. We could probably leverage K8up cli's restore command in the future.

How extensive should these tests be? Are they only for verifying that specific architectures were built correctly? E.g. does the binary work? Or do these tests need to be on a functional level? E.g. are there any bugs in the app?

For the first case I think that the `k8up --version` should be adequate
For the latter, we could incorporate the cli restore commands into the e2e tests

Best Regards

**Reply by kastl-ars** [reply]
> For the first case I think that the `k8up --version` should be adequate

Thanks for the reply, then I'll go with `k8up --version`!
